privaterelay/models.py

```diff
@@ -0,0 +1,561 @@
+                "addresses": abuse_metric.num_address_created_per_day,
+                "forwarded": abuse_metric.num_email_forwarded_per_day,
+                "forwarded_size_in_bytes": abuse_metric.forwarded_email_size_per_day,
+            }
+            # log for further secops review
+            abuse_logger.info("Abuse flagged", extra=data)
+        return self.last_account_flagged
+
+    @property
+    def is_flagged(self):
+        if not self.last_account_flagged:
+            return False
+        account_premium_feature_resumed = self.last_account_flagged + timedelta(
+            days=settings.PREMIUM_FEATURE_PAUSED_DAYS
+        )
+        if datetime.now(UTC) > account_premium_feature_resumed:
+            # premium feature has been resumed
+            return False
+        # user was flagged and the premium feature pause period is not yet over
+        return True
+
+    @property
+    def metrics_enabled(self) -> bool:
+        """
+        Does the user allow us to record technical and interaction data?
+
+        This is based on the Mozilla accounts opt-out option, added around 2022. A user
+        can go to their Mozilla account profile settings, Data Collection and Use, and
+        deselect "Help improve Mozilla Account". This setting defaults to On, and is
+        sent as "metricsEnabled". Some older Relay accounts do not have
+        "metricsEnabled", and we default to On.
+        """
+        if self.fxa:
+            return bool(self.fxa.extra_data.get("metricsEnabled", True))
+        return True
+
+    @property
+    def plan(self) -> Literal["free", "email", "phone", "bundle"]:
+        """The user's Relay plan as a string."""
+        if self.has_premium:
+            if self.has_phone:
+                return "bundle" if self.has_vpn else "phone"
+            else:
+                return "email"
+        else:
+            return "free"
+
+    @property
+    def plan_term(self) -> Literal[None, "unknown", "1_month", "1_year"]:
+        """The user's Relay plan term as a string."""
+        plan = self.plan
+        if plan == "free":
+            return None
+        if plan == "phone":
+            start_date = self.date_phone_subscription_start
+            end_date = self.date_phone_subscription_end
+            if start_date and end_date:
+                span = end_date - start_date
+                return "1_year" if span.days > 32 else "1_month"
+        return "unknown"
+
+    @property
+    def metrics_premium_status(self) -> str:
+        plan = self.plan
+        if plan == "free":
+            return "free"
+        return f"{plan}_{self.plan_term}"
+
+
+class RegisteredSubdomain(models.Model):
+    subdomain_hash = models.CharField(max_length=64, db_index=True, unique=True)
+    registered_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.subdomain_hash
+
+    class Meta:
+        db_table = "emails_registeredsubdomain"
```

Comment:
(suggestion, non-blocking) Add comment here that also points to migrations used and documentation on how the model was moved from `emails/models.py`